server/hap2/hatohol/rabbitmqconnector.py

```diff
@@ -134,15 +147,23 @@ def __publish(self, msg):
             logger.debug(msg)
             self.__publish_raw(msg)
         except:
-            # TODO: consider the way to save the message and try to
-            # send it again.
+            logger.error(traceback.format_exc())
             raise hap.Signal(critical=True)
 
     def __publish_raw(self, msg):
-        self._channel.basic_publish(exchange="", routing_key=self._queue_name,
-                                    body=msg,
-                                    properties=pika.BasicProperties(
-                                        content_type="application/json"))
+        publish_argument = {
+                            "exchange": "",
+                            "routing_key": self._queue_name,
+                            "body": msg,
+                            "properties": pika.BasicProperties(
+                                        content_type="application/json")
+                            }
+        try:
+            self._channel.basic_publish(**publish_argument)
+        except exceptions.ConnectionClosed:
+            self._connect()
```

Comment:
It seems that this retrying feature causes retrying when failing to publish forever.
Should we add upper limit of retrying count?
Or, this patch is enough for retrying feature for now?